File: C++/Model/equazione.cpp

#include "equazione.h"

equazione::equazione(const polinomio& s,const polinomio& d):sx(s),dx(d){}

polinomio equazione::unifica() const{
    return sx-dx;
}

string equazione::getEquazione() const{
    polinomio temp=unifica();
    temp.ordina();
    return temp.toString()+" = 0 ";
}

string equazione::sostituzioneX(const razionale& r){
    razionale a =sx.sostituisciVariabile(r);
    razionale b =dx.sostituisciVariabile(r);
    return a==b ? a.toString()+" = "+b.toString() : a.toString()+" != "+b.toString();
}

equazione::~equazione(){}

File: Java/equazione.java
public abstract class equazione{

    private polinomio sx,dx;

    public equazione(polinomio sx, polinomio dx){
        this.sx=sx;
        this.dx=dx;
    }

    public polinomio unifica(){
        try{
            return polinomio.sottrPolinomio(sx, dx);
        }
        catch(Exception e){
            return new polinomio();
        }
    }

    public String toString(){
        return unifica().toString()+" = 0";
    }

    public abstract String getRisultato();
    public abstract String getTipo();
    public abstract String getGrado();
    public abstract equazione getDerivata() throws erroreGrado;
    
}